Add `transitiveclosure` on SimpleGraphs

AsileBcd opened the pull request:
Add functions `transitiveclosure` and `transitiveclosure!` on SimpleGraphs, following issue #1114.
I hope this time I am not a lame duck.

AsileBcd commented on src/transitivity/transitivity.jl:
```diff
@@ -0,0 +1,30 @@
+Compute the transitive closure of an undirected graph.
+If `selflooped` is true, add self loops to the graph.
+"""
+function transitiveclosure(g::Graph; selflooped = false)
```
You are right, thanks!

AsileBcd commented on test/transitivity/transitivity.jl:
```diff
@@ -0,0 +1,30 @@
+            @test ne(circle) == 4
+            c = copy(circle)
+            @test newcircle == @inferred(transitiveclosure!(c))
+            @test ne(c) == ne(newcircle) == 12
```
Yes, 6, and the next one is 10, not 16.

AsileBcd commented on test/runtests.jl:
```diff
@@ -82,7 +82,8 @@ tests = [
     "vertexcover/random_vertex_cover",
-    "experimental/experimental"
+    "experimental/experimental",
+	"transitivity/transitivity"
```
It should be good now.

simonschoelly commented on src/LightGraphs.jl:
```diff
@@ -259,6 +259,7 @@ include("vertexcover/degree_vertex_cover.jl")
 include("vertexcover/random_vertex_cover.jl")
 include("Experimental/Experimental.jl")
 include("Parallel/Parallel.jl")
+include("transitivity/transitivity.jl")
```
If we have an extra file for transitivity, maybe we should also move `transitive_closure` and `transitive_reduction` for directed graphs to this file.

AsileBcd replied:
I just did not know where to put it, so I created a new file. If you have an idea where it should go, I definitely can move it here.
Otherwise, yes, I could do the reverse and bring DiGraph algorithms here.

simonschoelly commented on src/transitivity/transitivity.jl:
```diff
@@ -0,0 +1,30 @@
+### Implementation Notes
+This version of the function modifies the original graph.
+"""
+function transitiveclosure!(g::Graph, selflooped = false)
```
Maybe we should use `@traitfn` here - similar how we do it for directed graphs at the moment.

AsileBcd replied:
Sure!

simonschoelly commented on src/transitivity/transitivity.jl:
````diff
@@ -0,0 +1,30 @@
+"""
+    ```transitiveclosure!(g::SimpleGraph, selflooped = false)```
+Compute the transitive closure of an undirected graph.
+If `selflooped` is true, add self loops to the graph.
````
Maybe we should add to the documentation, that if the graph already has self-loops, but `selflooped` is set to `false`, then there self-loops will also be removed. We are currently not doing that for the directed version, but maybe we also should.

AsileBcd replied:
After thinking about it, I think that the current implementation is the right one. If one wants to modify the self loops, one should do it separately. The transitive closure is only taking care of edges _between_ different vertices, not _within_ the same vertex.

I will mention it in the docs.

simonschoelly commented on test/transitivity/transitivity.jl:
```diff
@@ -0,0 +1,30 @@
+@testset "Transitivity" begin
+    completeg = CompleteGraph(4)
+    circleg = PathGraph(4)
+    add_edge!(circleg, 4, 1)
```
You can use `CycleGraph(4)` here.

AsileBcd replied:
Thank you!

AsileBcd commented:
Thanks for the suggestions, I will come up with the updated modified version soon.
Your last points about tests is also true for the algorithm on DiGraphs, right?

AsileBcd commented:
I have modified the tests according to your suggestions, and went a bit beyond. I have removed the digraph folder and transferred everything regarding transitivity in the transitivity folder, and built two files, `transitiveclosure.jl` and `transitivereduction.jl`. I did the same for the tests.
I may be overtesting a bit, in the sense that I always test with and without self-loops, which might not be necessary. There is one thing I am not testing yet, but that I will add soon, which is that when `selfloop=false`, if there is self loops, they are removed.
